src/senselab/audio/tasks/forced_alignment/forced_alignment.py

```diff
@@ -0,0 +1,747 @@
+    model_type = align_model_metadata["type"]
+
+    transcript = _preprocess_segments(
+        transcript,
+        align_model_metadata["dictionary"],
+        align_model_metadata["language"],
+        print_progress,
+        combined_progress,
+    )
+
+    aligned_segments, word_segments = _align_segments(
+        transcript=transcript,
+        model=model,
+        model_dictionary=model_dictionary,
+        model_lang=model_lang,
+        model_type=model_type,
+        audio=audio,
+        device=device,
+        max_duration=max_duration,
+        return_char_alignments=return_char_alignments,
+        interpolate_method=interpolate_method,
+    )
+    return {"segments": aligned_segments, "word_segments": word_segments}
+
+
+def align_transcriptions(
+    audios_and_transcriptions: List[Tuple[Audio, ScriptLine]], language: Language = Language(language_code="en")
+) -> List[List[ScriptLine]]:
+    """Aligns transcriptions with the given audio using a wav2vec2.0 model.
+
+    Args:
+        audios_and_transcriptions (List[Tuple[Audio, ScriptLine]]):
+            A list of tuples where each tuple contains an audio object and its corresponding transcription.
+        language (Language): The language of the audio (default is "en").
+
+    Returns:
+        List[List[ScriptLine]]: A list of lists, where each inner list contains the aligned script lines for each audio.
+    """
+    aligned_script_lines = []
+
+    # Define the language code and load model
+    device = _select_device_and_dtype()[0]  # DeviceType object
+    model_name = DEFAULT_ALIGN_MODELS_HF.get(language.language_code, "facebook/wav2vec2-base-960h")
+
+    processor = Wav2Vec2Processor.from_pretrained(model_name)
+    model = Wav2Vec2ForCTC.from_pretrained(model_name).to(device.value)
+
+    for audio, transcription in audios_and_transcriptions:
+        audio = resample_audios([audio], SAMPLE_RATE)[0]  # Resample to fit expectations of Wav2Vec2
+
+        # Ensure start and end are not None
+        start = transcription.start if transcription.start is not None else 0.0
+        end = transcription.end if transcription.end is not None else audio.waveform.shape[1] / audio.sampling_rate
+
+        # Ensure text is not None
+        text = transcription.text if transcription.text is not None else ""
+
+        # Align each segment of the transcription
+        segments = [
+            SingleSegment(
+                start=start, end=end, text=text, clean_char=None, clean_cdx=None, clean_wdx=None, sentence_spans=None
+            )
+        ]
+
+        with threadpool_limits(limits=1, user_api="blas"):
+            alignment = _align_transcription(
+                transcript=segments,
+                model=model,
+                align_model_metadata={
+                    "dictionary": processor.tokenizer.get_vocab(),
+                    "language": language.language_code,
+                    "type": "huggingface",
```

Comment:
is this always "huggingface"? if you use the HFModel and TorchModel classes, you don't need this param anymore and your code looks prettier and is more readable

Reply:
ofc these changes would produce some cascade changes in the rest of the code

Reply:
Updated to use torchaudio when appropriate. Going to leave it the way it is for the sake of expediency...